server/src/main/java/org/elasticsearch/action/search/SearchRequestAttributesExtractor.java

```diff
@@ -266,8 +290,18 @@ private static void introspectQueryBuilder(QueryBuilder queryBuilder, QueryMetad
                 break;
             case RangeQueryBuilder range:
                 switch (range.fieldName()) {
-                    case TIMESTAMP -> queryMetadataBuilder.rangeOnTimestamp = true;
-                    case EVENT_INGESTED -> queryMetadataBuilder.rangeOnEventIngested = true;
+                    // don't track unbounded ranges, they translate to either match_none if the field does not exist
+                    // or match_all if the field is mapped
+                    case TIMESTAMP -> {
+                        if (range.to() != null || range.from() != null) {
```

Comment:
This is something I discovered as part of working on this chance: rewrite for range queries is a bit funky (I plan on fixing that as a follow-up). A range query may rewrite to one without bounds, if all data is within the range. In that case, it is equivalent to a match_all hence I don't report it as a range. I decided to rather report only "true" range queries at the shard level.